cms/djangoapps/contentstore/views/item.py

```diff
@@ -145,12 +146,19 @@ def xblock_handler(request, usage_key_string):
             accept_header = request.META.get('HTTP_ACCEPT', 'application/json')
 
             if 'application/json' in accept_header:
+                store = modulestore()
                 fields = request.GET.get('fields', '').split(',')
                 if 'graderType' in fields:
                     # right now can't combine output of this w/ output of _get_module_info, but worthy goal
                     return JsonResponse(CourseGradingModel.get_section_grader_type(usage_key))
+                if int(request.GET.get('course_tree', 0)):
+                    xblock = _get_xblock(usage_key, request.user)
+                    course = store.get_course(xblock.location.course_key)   # pylint: disable=no-member
+                    xblock_info = xblock_summary(xblock)
+                    course_outline = xblock_summary(course, include_children=True)
+                    return JsonResponse({'course_outline': course_outline, 'xblock_info': xblock_info})
```

Comment:
IMO it would make more sense to have this logic live in ``_get_module_info``, so why not pass that parameter to it. That would simplify this logic, and it would also have it wrapped inside ``bulk_operations``.